datalad/interface/common_opts.py

```diff
@@ -33,6 +33,13 @@
     constraints=EnsureInt() | EnsureNone(),
     doc="""limit recursion into subdataset to the given number of levels""")
 
+shared_access_opt = Parameter(
+    args=('--shared-access',),
```

Comment:
No no. I don't mind also renaming the current ``--shared`` to something more informative! ;-)